stable/victoria-metrics/templates/vminsert-deployment.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "victoria-metrics.fullname" . }}-vminsert
+  labels:
+    app.kubernetes.io/name: {{ include "victoria-metrics.fullname" . }}-vminsert
+    app.kubernetes.io/component: vminsert
+    {{ include "victoria-metrics.common-labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.vminsert.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "victoria-metrics.fullname" . }}-vminsert
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/component: vminsert
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "victoria-metrics.fullname" . }}-vminsert
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app.kubernetes.io/component: vminsert
+    spec:
+      terminationGracePeriodSeconds: {{ .Values.vminsert.terminationGracePeriodSeconds | int }}
+    {{- with .Values.vminsert.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      containers:
+        - name: vminsert
+          image: "{{ .Values.vminsert.image.repository }}:{{ .Values.vminsert.image.tag }}"
+          imagePullPolicy: {{ .Values.vminsert.image.pullPolicy }}
+          args:
+          {{- include "victoria-metrics.vminsert.vmstorage-pod-fqdn" . | nindent 12 }}
+          {{- range $key, $value := .Values.vminsert.extraArgs }}
+            - --{{ $key }}={{ $value }}
+          {{- end }}
+          ports:
+            - name: http
+              containerPort: 8480
+              protocol: TCP
+          readinessProbe:
+            httpGet:
+              path: /health
+              port: http
+            initialDelaySeconds: 5
+            periodSeconds: 15
+          resources:
+            {{- toYaml .Values.vminsert.resources | nindent 12 }}
+      {{- with .Values.vminsert.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
```

Comment:
It is simpler to write like this: `{{ toYaml .Values.vminsert.nodeSelector | indent 8 }}` instead of providing the context via `with`.